crates/astria-conductor/src/executor.rs

```diff
@@ -487,4 +505,34 @@ mod test {
         // should be empty because nothing was executed
         assert!(executor.sequencer_hash_to_execution_hash.is_empty());
     }
+
+    #[tokio::test]
+    async fn handle_block_received_from_data_availability_not_yet_executed_no_relevant_transactions_execute_block()
+     {
+        let conf = get_test_config();
+        let finalized_blocks = Arc::new(Mutex::new(HashSet::new()));
+        let execution_client = MockExecutionClient {
+            finalized_blocks: finalized_blocks.clone(),
+        };
+        let (mut executor, _) = Executor::new(execution_client, &conf).await.unwrap();
+
+        let block: SequencerBlockSubset = get_test_block_subset();
+        let expected_execution_state = hash(&executor.execution_state);
+
+        executor
+            .handle_block_received_from_data_availability(block)
+            .await
+            .unwrap();
+
+        assert!(
+            finalized_blocks
+                .lock()
+                .await
+                .get(&executor.execution_state)
+                .is_some()
+        );
+        assert_eq!(expected_execution_state, executor.execution_state,);
+        // should be empty because nothing was executed
+        assert!(executor.sequencer_hash_to_execution_hash.is_empty());
```

Comment:
is this correct? from the test name + previous lines i would expect something to be executed here

Reply:
The comment here was wrong, but the test is correct. The block is getting executed and finalized so it should not appear in the `sequencer_hash_to_exection_hash` map.